### PR TITLE
OHRM-174 : create plugin for marketplace and write test cases for APIs.

### PR DESCRIPTION
**Pull request by AshanKooragoda**

### CODE REVIEW

File: symfony/plugins/orangehrmMarketPlacePlugin/config/config.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+getAddons: 'https://jsonplaceholder.typicode.com/todos/'
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmMarketPlacePlugin/config/config.yml]: What is this external url?

File: symfony/plugins/orangehrmMarketPlacePlugin/config/orangehrmMarketPlacePluginConfiguration.class.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+{
+    /**
+     * @return bool|null|void
+     */
+    public function initialize()
+    {
+        $enabledModules = sfConfig::get('sf_enabled_modules');
+        if (is_array($enabledModules)) {
+            sfConfig::set('sf_enabled_modules', array_merge($enabledModules, array('maintenance')));
+//            untill verify keep MP under maintenance module
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmMarketPlacePlugin/config/orangehrmMarketPlacePluginConfiguration.class.php]: remove me

File: symfony/plugins/orangehrmMarketPlacePlugin/config/orangehrmMarketPlacePluginConfiguration.class.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ * Boston, MA 02110-1301, USA
+ */
+
+/**
+ * Class orangehrmMarketPlacePluginConfiguration
+ */
+class orangehrmMarketPlacePluginConfiguration extends sfPluginConfiguration
+{
+    /**
+     * @return bool|null|void
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmMarketPlacePlugin/config/orangehrmMarketPlacePluginConfiguration.class.php]: Is it return bool ?

File: symfony/plugins/orangehrmMarketPlacePlugin/modules/maintenance/actions/baseAddonAction.class.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
+<?php
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmMarketPlacePlugin/modules/maintenance/actions/baseAddonAction.class.php]: Why this class in maintenance module?

File: symfony/plugins/orangehrmMarketPlacePlugin/modules/maintenance/templates/ohrmAddonsSuccess.php [hunk 5]
```diff
@@ -0,0 +1,27 @@
+<?php
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmMarketPlacePlugin/modules/maintenance/templates/ohrmAddonsSuccess.php]: What is that ?

File: symfony/plugins/orangehrmMarketPlacePlugin/test/orangehrmMarketPlacePluginAllTests.php [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+/**
+ * Class orangehrmMarketPlacePluginAllTests
+ */
+class orangehrmMarketPlacePluginAllTests {
+
+    /**
+     * @return PHPUnit_Framework_TestSuite
+     */
+    public static function suite() {
+
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmMarketPlacePlugin/test/orangehrmMarketPlacePluginAllTests.php]: Why too much white space?

File: symfony/plugins/orangehrmMarketPlacePlugin/config/config.yml (hunk 1 above)
Comment (AshanKooragoda) [reply symfony/plugins/orangehrmMarketPlacePlugin/config/config.yml]: this is just to developing API request used for test purpose

File: symfony/plugins/orangehrmMarketPlacePlugin/config/orangehrmMarketPlacePluginConfiguration.class.php (hunk 2 above)
Comment (AshanKooragoda) [reply symfony/plugins/orangehrmMarketPlacePlugin/config/orangehrmMarketPlacePluginConfiguration.class.php]: what is the module name we go with?

File: symfony/plugins/orangehrmMarketPlacePlugin/config/orangehrmMarketPlacePluginConfiguration.class.php (hunk 3 above)
Comment (AshanKooragoda) [reply symfony/plugins/orangehrmMarketPlacePlugin/config/orangehrmMarketPlacePluginConfiguration.class.php]: yes it is auto generated doc comment.

**Pull request closed without merging**